storage-blobs-java-v10-tutorial-master/src/main/java/tutorial/Tutorial.java

```diff
@@ -49,109 +38,72 @@ static File createTempFile() throws IOException {
         return sampleFile;
     }
 
-    static void uploadFile(BlockBlobURL blob, File sourceFile) throws IOException {
-
-            AsynchronousFileChannel fileChannel = AsynchronousFileChannel.open(sourceFile.toPath());
-
-            // Uploading a file to the blobURL using the high-level methods available in TransferManager class
-            // Alternatively call the PutBlob/PutBlock low-level methods from BlockBlobURL type
-            TransferManager.uploadFileToBlockBlob(fileChannel, blob, 8*1024*1024, null)
-            .subscribe(response-> {
-                System.out.println("Completed upload request.");
-                System.out.println(response.response().statusCode());
-            });
-
-            
+    static void uploadFile(BlockBlobAsyncClient blob, File sourceFile) throws IOException {
+        ByteBuffer buffer = ByteBuffer.allocate(8 * 1024 * 1024);
+        AsynchronousFileChannel fileChannel = AsynchronousFileChannel.open(sourceFile.toPath());
+        fileChannel.read(buffer, 0);
+        blob.uploadWithResponse(new BlockBlobSimpleUploadOptions(
+                Flux.just(buffer), buffer.array().length
+        )).subscribe(response -> {
+            System.out.println("Completed upload request.");
+            System.out.println(response.getStatusCode());
+        });
     }
 
-    static void getBlob(BlockBlobURL blobURL, File sourceFile) {
-        // Get the blob using the low-level download method in BlockBlobURL type
-        // com.microsoft.rest.v2.util.FlowableUtil is a static class that contains helpers to work with Flowable
-        // BlobRange is defined from 0 to 4MB
-        blobURL.download(new BlobRange().withOffset(0).withCount(4*1024*1024L), null, false, null)
-                .flatMapCompletable(response -> {
-                    if(response.rawResponse().request().url().getHost().contains("-secondary"))
-                    {
+    static void getBlob(BlockBlobAsyncClient blockBlobAsyncClient, File sourceFile) {
+        blockBlobAsyncClient.downloadStreamWithResponse(
+                new BlobRange(0, 4 * 1024 * 1024L),
+                null, null, false)
+                .doOnSubscribe(onSubscribe -> System.out.println("The blob was downloaded to " + sourceFile.getAbsolutePath()))
```

Comment:
`doOnSubscribe` occurs before the file is downloaded, it occurs once the download operation begins, let's reword this to:

```suggestion
                .doOnSubscribe(onSubscribe -> System.out.println("The blob is being downloaded to " + sourceFile.getAbsolutePath()))
```